website/static/js/filepage/revisions.js

```diff
@@ -218,7 +218,12 @@ var FileRevisionsTable = {
         }
 
         if (file.provider === 'osfstorage' && file.name && index !== 0) {
-            var parts = file.name.split('.');
+            var parts;
+            if (file.versionNames && file.versionNames.length) {
+                parts = file.versionNames[index].split('.');
+            } else {
+                parts = file.name.split('.');
+            }
```

Comment:
Legacy file revision download name is set here.  It was pulling the name off the file.  The loaded file revision comes from a WB request, which currently doesn't have the per version name on it.  To avoid a third request to get versions from the OSF, I'm just adding a list of `version_names` to the serialized file.  


With this change, if older versions had different names, that is part of the base name. I didn't touch the portion that adds the modified date to older versions. 

In this example, version 1 is named first_name, version 2 is named second_name, and version 3 (latest) is named third_name -
![legacy_version_download](https://user-images.githubusercontent.com/9755598/60462364-b31ce000-9c0e-11e9-96a1-2cfe3f78481b.gif)